packages/invoices/src/OutboundShipment/DetailView/tabs/GeneralTab.tsx

```diff
@@ -38,22 +80,80 @@ export const GeneralTabComponent: FC<
   GeneralTabProps<OutboundShipmentSummaryItem>
 > = ({ data, columns, onRowClick }) => {
   const { pagination } = usePagination();
-  const t = useTranslation('common');
-  const activeRows = data.filter(({ isDeleted }) => !isDeleted);
+  const { isOn, toggle } = useToggle();
+
+  const [grouped, setGrouped] = React.useState<OutboundShipmentSummaryItem[]>(
+    []
+  );
+  const paged = data.slice(
+    pagination.offset,
+    pagination.offset + pagination.first
+  );
+
+  React.useEffect(() => {
+    const newGrouped: OutboundShipmentSummaryItem[] = [];
+    paged.forEach(row => {
+      const batches = Object.values(row.batches);
+      const lineTotal = (row.sellPricePerPack ?? 0) * (row.numberOfPacks ?? 0);
+
+      if (isOn) {
+        newGrouped.push({
+          ...row,
+          lineTotal,
+          sellPricePerUnit: lineTotal / row.unitQuantity,
+          batch: ifTheSameElseDefault(batches, 'batch', '[multiple]'),
+          expiryDate: ifTheSameElseDefault(batches, 'expiryDate', '[multiple]'),
+          canExpand: Object.keys(row.batches).length > 1,
+        });
+      } else {
+        batches.forEach(batch => {
+          newGrouped.push({
+            ...row,
+            batch: batch.batch,
+            expiryDate: batch.expiryDate,
+            packSize: batch.packSize,
+            lineTotal,
+            sellPricePerUnit: lineTotal / row.unitQuantity,
```

Comment:
This `lineTotal` is for all the lines of the item, rather than this specific line - is that intentional? 

Actually I think maybe the `lineTotal` column should just use an `accessor` function rather than trying to keep this field up to date? 

Should the `sellPricePerUnit` be coming from the API? 

I would have thought the `sellPricePerPack` would be more useful than the `sellPricePerUnit` which we aren't displaying?

Reply:
Hm that's weird .. wouldn't really expect anything other than the line total in the summary row!